lib/LANraragi/Utils/Archive.pm

```diff
@@ -44,44 +46,14 @@ sub is_pdf {
 # If use_jxl is true, JPEG XL will be used instead of JPEG.
 sub generate_thumbnail ( $data, $thumb_path, $use_hq, $use_jxl ) {
 
-    no warnings 'experimental::try';
-    my $img = undef;
-    try {
-        require Image::Magick;
-        $img = Image::Magick->new;
-
-        my $format = $use_jxl ? 'jxl' : 'jpg';
-
-        # For JPEG, the size option (or jpeg:size option) provides a hint to the JPEG decoder
-        # that it can reduce the size on-the-fly during decoding. This saves memory because
-        # it never has to allocate memory for the full-sized image
-        if ( $format eq 'jpg' ) {
-            $img->Set( option => 'jpeg:size=500x' );
-        }
-
-        $img->BlobToImage($data);
-
-        # Only use the first frame (relevant for animated gif/webp/whatever)
-        $img = $img->[0];
-
-        # The "-scale" resize operator is a simplified, faster form of the resize command.
-        if ($use_hq) {
-            $img->Scale( geometry => '500x1000' );
-        } else {    # Sample is very fast due to not applying filters.
-            $img->Sample( geometry => '500x1000' );
-        }
-
-        $img->Set( quality => "50", magick => $format );
-        $img->Write($thumb_path);
-    } catch ($e) {
-
-        # Magick is unavailable, do nothing
+    my $resized = get_resizer()->resize_thumbnail( $data, 80, $use_hq, $use_jxl?"jxl":"jpg");
```

Comment:
It's probably fine to remove $quality from the `resize_thumbnail` API and just hardcode it 80 in the `ImageMagickResizer` if it's unnecessary for vips.  

It might be worth amending the config page a bit to mention that the Quality parameter for resized pages won't do anything if VIPS is installed too!  
(I'm almost considering we could remove it entirely, but that might be better for a second iteration/release with this stuff in)